automata/fa/dfa.py

```diff
@@ -606,6 +718,11 @@ def from_finite_language(cls, language, input_symbols):
         Chapter 10
         """
 
+        if len(language) == 0:
```

Comment:
Can you have this throw an exception instead? I want to write a class method that returns the empty DFA later.

Similarly, I think this algorithm doesn't work with the empty string, so it should throw an exception in that case.

Reply:
One of my tests for len was using the function to create the empty language, which is finite, and therefore should be possible to create with this function in my opinion. I don't think we should be raising an exception in that case.

The algorithm does work with the empty string, at least what I've tried so far seems to work.

Reply:
However I do agree with adding a specific function for creating the empty language and this function can then defer to that function for this edge case.

Reply:
Sounds good. Did you try the empty string with at least one other string? I think that might be the situation that causes problems.

Reply:
I tried the language with only the empty word and then two other finite languages that were larger but also included the empty word.